src/ConnectedMode.UnitTests/Migration/MigrationCheckerTests.cs

```diff
@@ -53,20 +55,37 @@ public void CheckIsSingletonMefComponent()
         [DataRow(SonarLintMode.LegacyConnected, SonarLintMode.Standalone, true)]
         [DataRow(SonarLintMode.Connected, SonarLintMode.Connected, false)]
         [DataRow(SonarLintMode.LegacyConnected, SonarLintMode.Connected, false)]
-        public async Task Migrate_BindingGetsCalledWithCorrectCondition(SonarLintMode obsoleteMode, SonarLintMode mode, bool expectBindingToBeCalled)
+        [DataRow(null, SonarLintMode.Connected, false)]
+        [DataRow(SonarLintMode.LegacyConnected, null, false)]
```

Comment:
In an earlier version of this PR, an analysis bug was reported complaining that the old binding config could be `null`. I fixed the product code and modified the test setup so it can handle the binding configs being `null`.